src/server.rs

```diff
@@ -65,11 +64,8 @@ impl Server {
     }
 
     fn handle_request(&self, stream: TcpStream) -> Result<()> {
-        let reader = stream.try_clone()?;
-        //discard because: https://doc.rust-lang.org/std/net/struct.TcpStream.html#method.set_read_timeout
-        let _ = reader.set_read_timeout(Some(Duration::from_millis(1000)));
```

Comment:
I don't see a problem with keeping a timeout regardless of the other code. (Maybe 1 second is too small, but in reality I don't think it'll be a problem.)

Reply:
I'll add it back in, honestly, I thought 1 second was too large.